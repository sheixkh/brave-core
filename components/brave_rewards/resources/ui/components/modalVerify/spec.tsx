/* global jest, expect, describe, it, afterEach */
import * as React from 'react'
import { shallow } from 'enzyme'
import { create } from 'react-test-renderer'
import PanelVerify from './index'
import { TestThemeProvider } from 'brave-ui/theme'

describe('PanelVerify tests', () => {
  const baseComponent = (props?: object) => <TestThemeProvider><PanelVerify id='panel-verify' {...props} /></TestThemeProvider>

  describe('basic tests', () => {
    it('matches the snapshot', () => {
      const component = baseComponent()
      const tree = create(component).toJSON()
      expect(tree).toMatchSnapshot()
    })

    it('renders the component', () => {
      const wrapper = shallow(baseComponent())
      const assertion = wrapper.find('#panel-verify').length
      expect(assertion).toBe(1)
    })
  })
})
