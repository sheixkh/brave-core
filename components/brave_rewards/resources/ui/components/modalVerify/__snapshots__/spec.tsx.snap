// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PanelVerify tests basic tests matches the snapshot 1`] = `
<div
  className="sc-kxynE lkjYuP"
  id="panel-verify"
  onClick={[Function]}
>
  <div
    className="sc-cooIXK hiIIWL"
    size="small"
  >
    <div
      className="sc-gmeYpB cpHCod"
    >
      <div
        className="sc-chPdSV eaFjmb"
        id="panel-verify"
      >
        <div
          className="sc-kgoBCf hTipRj"
        >
          <svg
            aria-hidden="true"
            className="sc-ktHwxA dGfSmA"
            focusable="false"
            viewBox="0 0 32 32"
          >
            <path
              d="M28.71 27.29L17.41 16l11.3-11.29a1 1 0 1 0-1.42-1.42L16 14.59 4.71 3.29a1 1 0 0 0-1.42 1.42L14.59 16 3.29 27.29a1 1 0 0 0 0 1.42 1 1 0 0 0 1.42 0L16 17.41l11.29 11.3a1 1 0 0 0 1.42 0 1 1 0 0 0 0-1.42z"
            />
          </svg>
        </div>
        <div
          className="sc-kGXeez jktXSx"
        >
          <div
            className="sc-kpOJdX bvxOSq"
          >
            <svg
              aria-hidden="true"
              className="sc-CtfFt jvAZOy"
              focusable="false"
              viewBox="0 0 32 32"
            >
              <defs>
                <path
                  d="M.003.032h23.21V23H.002z"
                  id="rewards-a"
                />
              </defs>
              <g
                fill="none"
                fillRule="evenodd"
              >
                <g
                  transform="translate(6.755 4.5)"
                >
                  <mask
                    fill="#fff"
                    id="rewards-b"
                  >
                    <use
                      xlinkHref="#rewards-a"
                    />
                  </mask>
                  <path
                    d="M22.157 16.214a.525.525 0 0 1-.527.522h-3.693a2.1 2.1 0 0 1-2.11-2.088c0-1.151.946-2.088 2.11-2.088h3.693c.29 0 .527.234.527.522v3.132zm-2.11 5.742H2.11a1.051 1.051 0 0 1-1.055-1.044V8.384c0-.575.474-1.044 1.055-1.044h17.937c.582 0 1.055.469 1.055 1.044v3.132h-3.165c-1.746 0-3.166 1.405-3.166 3.132s1.42 3.132 3.166 3.132h3.165v3.132a1.05 1.05 0 0 1-1.055 1.044zm-5.195-20.74a1.03 1.03 0 0 1 .783-.105c.27.072.495.244.634.485l.001.002-8.156 4.698H6.031l8.82-5.08zm2.987 3.094l1.15 1.986H14.39l3.448-1.986zm-7.628 1.986l6.583-3.792.521.9-5.021 2.892H10.21zm11.946 5.316V8.384c0-1.094-.858-1.984-1.942-2.07l-3.03-5.237a2.082 2.082 0 0 0-1.277-.974 2.092 2.092 0 0 0-1.587.21L3.934 6.296H2.11C.946 6.296 0 7.233 0 8.384v12.528A2.1 2.1 0 0 0 2.11 23h17.937a2.1 2.1 0 0 0 2.11-2.088v-3.228a1.565 1.565 0 0 0 1.055-1.47v-3.132c0-.68-.442-1.253-1.055-1.47z"
                    fill="#A1A8F2"
                    mask="url(#rewards-b)"
                  />
                </g>
                <path
                  d="M24.132 18.176a.939.939 0 0 0-.943.932c0 .514.423.933.943.933s.942-.419.942-.933a.939.939 0 0 0-.942-.932"
                  fill="#A1A8F2"
                />
                <g
                  transform="translate(2 8.682)"
                >
                  <ellipse
                    cx="6.34"
                    cy="6.273"
                    fill="#1BBA6A"
                    rx="6.34"
                    ry="6.273"
                  />
                  <path
                    d="M2.635 6.502l.462-.729 2.583 1.51 3.968-3.307.517.694L5.68 9.827z"
                    fill="#FFF"
                  />
                </g>
              </g>
            </svg>
          </div>
          <div
            className="sc-dxgOiQ gUVnKj"
          >
            <h1
              className="sc-ckVGcZ jxMEaM sc-htpNat gMpIXU sc-bwzfXH fYPHla"
            >
              MISSING: walletVerificationTitle1
            </h1>
            <h2
              className="sc-jKJlTe eRgNpf sc-ifAKCX bWuOYq sc-bxivhb gycNlz"
            >
              MISSING: walletVerificationTitle2
            </h2>
          </div>
        </div>
        <h1
          className="sc-eNQAEJ DPpYL sc-htpNat gMpIXU sc-bwzfXH fYPHla"
        >
          MISSING: walletVerificationListHeader
        </h1>
        <div
          className="sc-cvbbAY dZpbKN"
        >
          <div
            className="sc-jWBwVP iObYtR"
          >
            <div
              className="sc-hMqMXs fteRZz"
            >
              <div
                className="sc-kEYyzF jvSIwV"
              >
                <svg
                  aria-hidden="true"
                  className="sc-rBLzX gepCvW"
                  focusable="false"
                  viewBox="0 0 32 32"
                >
                  <g
                    fill="none"
                    fillRule="evenodd"
                    transform="translate(2 2)"
                  >
                    <circle
                      cx="14"
                      cy="14"
                      fill="#FFF"
                      r="14"
                    />
                    <path
                      d="M5.818 14.512l1.02-1.627 5.704 3.37 8.764-7.381 1.143 1.548-9.907 11.511z"
                      fill="#A1A8F2"
                    />
                  </g>
                </svg>
              </div>
              <div
                className="sc-kkGfuU cgtoYx"
              >
                MISSING: walletVerificationList1
              </div>
            </div>
            <div
              className="sc-hMqMXs fteRZz"
            >
              <div
                className="sc-kEYyzF jvSIwV"
              >
                <svg
                  aria-hidden="true"
                  className="sc-rBLzX gepCvW"
                  focusable="false"
                  viewBox="0 0 32 32"
                >
                  <g
                    fill="none"
                    fillRule="evenodd"
                    transform="translate(2 2)"
                  >
                    <circle
                      cx="14"
                      cy="14"
                      fill="#FFF"
                      r="14"
                    />
                    <path
                      d="M5.818 14.512l1.02-1.627 5.704 3.37 8.764-7.381 1.143 1.548-9.907 11.511z"
                      fill="#A1A8F2"
                    />
                  </g>
                </svg>
              </div>
              <div
                className="sc-kkGfuU cgtoYx"
              >
                MISSING: walletVerificationList2
              </div>
            </div>
            <div
              className="sc-hMqMXs fteRZz"
            >
              <div
                className="sc-kEYyzF jvSIwV"
              >
                <svg
                  aria-hidden="true"
                  className="sc-rBLzX gepCvW"
                  focusable="false"
                  viewBox="0 0 32 32"
                >
                  <g
                    fill="none"
                    fillRule="evenodd"
                    transform="translate(2 2)"
                  >
                    <circle
                      cx="14"
                      cy="14"
                      fill="#FFF"
                      r="14"
                    />
                    <path
                      d="M5.818 14.512l1.02-1.627 5.704 3.37 8.764-7.381 1.143 1.548-9.907 11.511z"
                      fill="#A1A8F2"
                    />
                  </g>
                </svg>
              </div>
              <div
                className="sc-kkGfuU cgtoYx"
              >
                MISSING: walletVerificationList3
              </div>
            </div>
          </div>
          <div
            className="sc-brqgnP gEfbbI"
          >
            <button
              className="sc-iAyFgw fxlfDk sc-jTzLTM msywM sc-VigVT iFaqXL"
              id="on-boarding-verify-button"
              size="call-to-action"
              type="accent"
            >
              <div
                className="sc-cSHVUG ckJrzp"
                size="call-to-action"
                type="accent"
              >
                MISSING: walletVerificationButton
              </div>
            </button>
            <div
              className="sc-hSdWYo bhJSll"
            >
              MISSING: walletVerificationFooter
               
              <b>
                Uphold
              </b>
              <div
                className="sc-eHgmQL gDMUwd"
              >
                <svg
                  aria-hidden="true"
                  className="sc-dliRfk jiqQOG"
                  focusable="false"
                  viewBox="0 0 32 32"
                >
                  <path
                    d="M18.6,27.4c-0.8,0.2-1.6,0.4-2.4,0.4h-0.1c-0.8,0-1.6-0.1-2.4-0.4c-0.4-0.1-0.8,0.1-0.9,0.5s0.1,0.8,0.5,0.9 c0.9,0.3,1.9,0.4,2.8,0.4h0.1c1,0,1.9-0.1,2.8-0.4c0.4-0.1,0.6-0.5,0.5-0.9C19.4,27.5,19,27.2,18.6,27.4z M24.9,9.6L24.9,9.6 L24.9,9.6C23.3,4.7,18.1,2,13.3,3.6c-2.8,0.9-5,3.1-5.9,5.9l0,0c-0.8,2.4-0.5,5.6,0.8,8.6c1.8,4.1,5.1,6.9,8,6.9h0.1 c2.9,0,6.2-2.8,8-6.9C25.4,15.2,25.7,12,24.9,9.6z M20.7,6c-1.7,0-3.3,0.6-4.5,1.8c-1.2-1.1-2.9-1.8-4.5-1.8 C14.3,4.1,17.9,4.1,20.7,6z M17.3,23.4c-0.8,0.4-1.6,0.4-2.4,0c-2.4-1.1-3-5.7-1.1-10c0.6-1.3,1.3-2.5,2.3-3.6 c1,1.1,1.7,2.3,2.3,3.6C20.3,17.7,19.8,22.3,17.3,23.4z M9.4,17.6C8.3,14.9,8,12.1,8.7,10C9,9,9.7,8.2,10.5,7.8 c1.3-0.6,2.9-0.2,4.5,1.1c-1.1,1.2-1.9,2.5-2.6,4c-1.2,2.5-1.5,5.4-1,8.1C10.7,20,10,18.8,9.4,17.6z M22.8,17.6 c-0.5,1.2-1.2,2.4-2.1,3.4c0.6-2.7,0.2-5.6-1-8.1c-0.6-1.5-1.5-2.8-2.6-4c1.6-1.3,3.3-1.7,4.5-1.1c0.9,0.5,1.6,1.3,1.8,2.2 C24.3,12.1,24,14.9,22.8,17.6z"
                    fill="#49cc68"
                  />
                </svg>
              </div>
            </div>
            <div
              className="sc-cMljjf bWWeVK"
            >
              MISSING: walletVerificationNote1
              <br />
              <br />
              MISSING: walletVerificationNote2
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
